Add in Jira/Ticket reference to commit-msg

davidcollom opened the pull request:
Add in `commit-msg` style hooks along with a jira/ticket reference hook

greg0ire commented:
Looks great!

Can you make it more generic though? Like, not tied to jira? And please add some documentation describing what this does.

greg0ire commented:
Also, what happens/should happen when you have several commits on the branch?

davidcollom commented:
Hey, I've squashed my commits and removed references of Jira into a `ticketref`plugin.

The expected behavior of this plugin is to prefix the commit message with the ticket reference from the branch name to ensure that the correlation of commit's against a specific feature/ticket/incident/issue etc.

greg0ire commented on docs/hooks/ticketref.md:
```diff
@@ -0,0 +1,11 @@
+# Description
+
+Prefixes the commit-msg with the ticket reference, this allows for some workflows to be able to correlate a number of commit's against a specific feature/ticket/incident/issue etc once merged.
```
- Please wrap after 80 chars (`gqq` in vim)
- "commit's" => "commits"

greg0ire replied:
Also please add a dot after `etc`, it's an abbreviation.

greg0ire commented on docs/hooks/ticketref.md:
```diff
@@ -0,0 +1,11 @@
+
+Prefixes the commit-msg with the ticket reference, this allows for some workflows to be able to correlate a number of commit's against a specific feature/ticket/incident/issue etc once merged.
+
+a branch of `ABC-1234-hello-world` would turn a commit of `git commit -m "Hello World"` into `[ABC-1234] Hello World`
```
Please capitalize the first word and put a dot at end of sentence.

greg0ire replied:
This has the big drawback of making the first line longer. The first line is supposed to be less than 50 chars. Would it be possible to produce something like this instead:

```
Hello world

Refs ABC-1234
```

greg0ire commented:
cc @juliendufresne :P

Edo78 commented:
This looks interesting for my use case. Any chance it could be updated and accepted?

greg0ire commented:
@Edo78 sure! You can make your own PR and address the issues, I will merge it!

Edo78 commented:
I feel a bit uneasy at copying the code of @davidcollom 
If he could give me his permission I'll be glad to do it

davidcollom commented:
Hey, really sorry I completely forgot about the PR.

I may have some time today to update the PR based off the comments above, I don't see them taking much time.

davidcollom commented:
@Edo78 / @greg0ire - I've updated this PR based off the comments you left prior

I added some additional configuration for this hook to support both requirements (Pre/Post reference locations).
